build_runner/lib/src/build_script_generate/bootstrap.dart

```diff
@@ -187,13 +189,19 @@ final _previousLocationsFile = File(
 /// pre-emptively resolve them by resnapshotting, see
 /// https://github.com/dart-lang/build/issues/1929.
 Future<bool> _checkImportantPackageDeps() async {
-  var currentLocationsContent = _importantPackages
-      .map((pkg) => Isolate.resolvePackageUri(
+  var currentLocationsContent = await Stream.fromIterable(_importantPackages)
+      .asyncMap((pkg) => Isolate.resolvePackageUri(
```

Comment:
There wasn't any existing testing around this content, so I wasn't sure what to do.

If you'd like me to add a test, just let me know how I should go about this.